### PR TITLE
ref(react-native): Move navigation guides to new pages per supported library

### PR DESCRIPTION
**Pull request by krystofwoldrich**
No new content, just moved the guide to new pages per supported library, to make it easier to consume.

### CODE REVIEW

File: docs/platforms/react-native/performance/instrumentation/react-native-navigation.mdx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+```javascript
+new Sentry.ReactNativeNavigationInstrumentation({
+  routeChangeTimeoutMs: 1000, // default: 1000
+});
+```
+
+### routeChangeTimeoutMs
+
+How long the instrumentation will wait for the route to mount after a change has been initiated, before the transaction is discarded.
```
Comment (vivianyentran) [inline docs/platforms/react-native/performance/instrumentation/react-native-navigation.mdx:43]:
```suggestion
This option specifies how long the instrumentation will wait for the route to mount after a change has been initiated before the transaction is discarded. The default value is 1000ms.
```

File: docs/platforms/react-native/performance/instrumentation/react-navigation-v4.mdx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+```javascript
+new Sentry.ReactNavigationV4Instrumentation({
+  routeChangeTimeoutMs: 1000, // default: 1000
+});
+```
+
+### routeChangeTimeoutMs
+
+How long the instrumentation will wait for the **initial** route to mount after a change has been initiated before the transaction is discarded. Default: 1000ms
```
Comment (vivianyentran) [inline docs/platforms/react-native/performance/instrumentation/react-navigation-v4.mdx:86]:
```suggestion
This option specifies how long the instrumentation will wait for the **initial** route to mount after a change has been initiated before the transaction is discarded. The default value is 1000ms.
```

File: docs/platforms/react-native/performance/instrumentation/react-navigation.mdx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+```javascript
+new Sentry.ReactNavigationInstrumentation({
+  enableTimeToInitialDisplay: true, // default: false
+  routeChangeTimeoutMs: 1000, // default: 1000
+});
+```
+
+### routeChangeTimeoutMs
+
+How long the instrumentation will wait for the route to mount after a change has been initiated before the transaction is discarded. Default: 1000ms
```
Comment (vivianyentran) [inline docs/platforms/react-native/performance/instrumentation/react-navigation.mdx:28]:
```suggestion
This option specifies how long the instrumentation will wait for the route to mount after a change has been initiated before the transaction is discarded. The default value is 1000ms.
```

File: docs/platforms/react-native/performance/instrumentation/react-navigation.mdx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+### enableTimeToInitialDisplay
+
+This option will enabled automatic measuring og the time to initial display for each route. To learn more see [Time to Initial Display](/platforms/react-native/performance/instrumentation/time-to-display).
```
Comment (vivianyentran) [inline docs/platforms/react-native/performance/instrumentation/react-navigation.mdx:32]:
```suggestion
This option will enable automatic measuring of the time to initial display for each route. To learn more see [Time to Initial Display](/platforms/react-native/performance/instrumentation/time-to-display).
```

File: docs/platforms/react-native/performance/instrumentation/react-navigation.mdx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+## Notes
+
+- This instrumentation supports React Navigation version 5 and above. If you use React Navigation version 4, see the [instrumentation for React Navigation V4](/platforms/react-native/performance/instrumentation/react-navigation-v4/).
+
+- The instrumentation creates a transaction on every route change including `goBack` navigations.
+
+- If you are coming from a version prior to 2.3.0, make sure you update where `registerNavigationContainer` is called. You can find the instructions on [our migration guide](/platforms/react-native/migration/#react-navigation-instrumentation-from-230).
```
Comment (vivianyentran) [inline docs/platforms/react-native/performance/instrumentation/react-navigation.mdx:40]:
```suggestion
- If you are coming from a version prior to 2.3.0, make sure you update where `registerNavigationContainer` is called. For more detailed instructions, see [our migration guide](/platforms/react-native/migration/#react-navigation-instrumentation-from-230).
```

File: docs/platforms/react-native/performance/instrumentation/react-navigation-v4.mdx [hunk 6]
```diff
@@ -0,0 +1,92 @@
+---
+title: React Navigation V4
+description: "Learn how to use Sentry's React Navigation Legacy instrumentation."
+sidebar_order: 51
+---
+
+Sentry's React Native SDK package ships with instrumentation for React Navigation V4 and older. This allows you to see the performance of your navigation transitions and the errors that occur during them. This page will guide you through setting up the instrumentation and configuring it to your needs.
+
```
Comment (vivianyentran) [inline docs/platforms/react-native/performance/instrumentation/react-navigation-v4.mdx:8]:
```suggestion

## Initialization

The code snippet below shows how to initialize the instrumentation.

```
Adding this section header for consistency with the other pages

File: docs/platforms/react-native/performance/instrumentation/automatic-instrumentation.mdx [hunk 7]
```diff
@@ -28,200 +28,13 @@ When no routing instrumentation is used, a transaction for `App Start` is automa
 
 We currently provide three routing instrumentations out of the box to instrument route changes for:
 
-- [React Navigation](#react-navigation)
-- [React Navigation V4 and prior](#react-navigation-v4)
-- [React Native Navigation](#react-native-navigation)
+- [React Navigation](/platforms/react-native/performance/instrumentation/react-navigation/)
+- [React Navigation V4 and prior](/platforms/react-native/performance/instrumentation/react-navigation-v4/)
+- [React Native Navigation](/platforms/react-native/performance/instrumentation/react-native-navigation/)
+- [Custom Instrumentation](/platforms/react-native/performance/instrumentation/custom-navigation/), implement automatic navigation instrumentation for a custom library.
```
[the remark this answers is not in the document]
Comment (vivianyentran) [reply docs/platforms/react-native/performance/instrumentation/automatic-instrumentation.mdx:34]: Remember to add expo router to this list when it gets published :)

**Review by vivianyentran — APPROVED** [review]
Added some small edits, but otherwise LGTM. Thanks for restructuring these docs!

File: docs/platforms/react-native/performance/instrumentation/automatic-instrumentation.mdx (hunk 7 above)
[the remark this answers is not in the document]
Comment (krystofwoldrich) [reply docs/platforms/react-native/performance/instrumentation/automatic-instrumentation.mdx:34]: Yes, definitely, here it is https://github.com/getsentry/sentry-docs/pull/9501/files#diff-7364040069f17041d27e5d8716f6f8e6e170fc8d9062368f18f91e772a42a909

**Pull request merged**